tests/plugin_tests/follow.cpp

```diff
@@ -0,0 +1,202 @@
+#include <boost/test/unit_test.hpp>
+
+#include <boost/container/flat_set.hpp>
+
+#include "database_fixture.hpp"
+#include "helpers.hpp"
+
+#include <golos/chain/account_object.hpp>
+
+#include <golos/plugins/follow/plugin.hpp>
+#include <golos/plugins/follow/follow_operations.hpp>
+
+using boost::container::flat_set;
+
+using golos::plugins::json_rpc::msg_pack;
+
+using golos::logic_exception;
+using golos::missing_object;
+using golos::protocol::comment_operation;
+using golos::protocol::tx_invalid_operation;
+using golos::protocol::public_key_type;
+using golos::protocol::signed_transaction;
+using golos::protocol::custom_binary_operation;
+using golos::chain::account_id_type;
+using golos::chain::make_comment_id;
+
+using namespace golos::plugins::follow;
+
+using account_name_set = flat_set<account_name_type>;
+
+
+struct follow_fixture : public golos::chain::database_fixture {
+    follow_fixture() : golos::chain::database_fixture() {
+        initialize<golos::plugins::follow::plugin>();
+        open_database();
+        startup();
+    }
+};
+
+
+BOOST_FIXTURE_TEST_SUITE(follow_plugin, follow_fixture)
+
+BOOST_AUTO_TEST_CASE(follow_validate) {
+    BOOST_TEST_MESSAGE("Testing: follow_validate");
+
+    ACTORS((alice)(bob));
+
+    follow_operation op;
+    op.follower = "alice";
+    op.following = "bob";
+    op.what = {"blog"};
+    CHECK_OP_VALID(op);
+
+    CHECK_PARAM_VALIDATION_FAIL(op, following, "alice",
+        CHECK_ERROR(logic_exception, logic_errors::cannot_follow_yourself));
+}
+
+BOOST_AUTO_TEST_CASE(follow_authorities) {
+    BOOST_TEST_MESSAGE("Testing: follow_authorities");
+
+    follow_operation op;
+    op.follower = "alice";
+    op.following = "bob";
+    op.what = {"blog"};
+
+    CHECK_OP_AUTHS(op, account_name_set(), account_name_set(), account_name_set({"alice"}));
+}
+
+BOOST_AUTO_TEST_CASE(follow_apply) {
+    BOOST_TEST_MESSAGE("Testing: follow_apply");
+
+    ACTORS((alice)(bob));
+
+    generate_blocks(60 / STEEMIT_BLOCK_INTERVAL);
+
+    custom_binary_operation cop;
```

Comment:
Maybe we should test `custom_json_operation` path too. At least in one plugin, which use custom operations.

Reply:
Ideally, it's must be done in tests for custom_json_operations.